assets/js/amazon-wc-checkout.js

```diff
@@ -5,11 +5,13 @@
 		var button_id = '#pay_with_amazon';
 		var classicButtonId = '#classic_pay_with_amazon';
 		var amazonCreateCheckoutConfig = null;
+		var amazonEstimatedOrderAmount = null;
 
 		/* Handles 'classic' payment method on checkout. */
 		$( 'form.checkout' ).on( 'checkout_place_order_success', function( e, result ) {
 			if ( 'undefined' !== typeof result.amazonCreateCheckoutParams && $( classicButtonId ).length > 0 ) {
 				amazonCreateCheckoutConfig = JSON.parse( result.amazonCreateCheckoutParams );
+				amazonEstimatedOrderAmount = 'undefined' !== typeof result.amazonEstimatedOrderAmount && result.amazonEstimatedOrderAmount ? JSON.parse( result.amazonEstimatedOrderAmount ) : null;
```

Comment:
Would `JSON.parse( result.amazonEstimatedOrderAmount || 'null' )` behave the same as `undefined' !== typeof result.amazonEstimatedOrderAmount && result.amazonEstimatedOrderAmount ? JSON.parse( result.amazonEstimatedOrderAmount ) : null`? Or it's possible that semantic changes?

Reply:
Hmm, im a bit worried of the possibility when `result.amazonEstimatedOrderAmount` is undefined that trying to access its value could produce a warning or an error. In modern JS, it would definitely be possible since we could
`JSON.parse( result?.amazonEstimatedOrderAmount || null )`

Did a test in my browser's console though (chrome) and there was no warning. The test was:

```
var result = {};
console.log( JSON.parse( result.test || null ) );
```

The output was simply `null` without a warning or anything. To be honest, even though the code suggestion is really good, i feel like it would be better to introduce it in a later version where we will have the time to test it properly.